clippy_lints/src/transmute.rs

```diff
@@ -630,6 +630,25 @@ impl<'tcx> LateLintPass<'tcx> for Transmute {
                             );
                         }
                     },
+                    (ty::FnDef(..), ty::Int(_) | ty::Uint(_)) => span_lint_and_then(
```

Comment:
~~This is unrelated?~~

I wouldn't do this, since the extra cast to `fn() -> T` is code bloat. If people prefer these explicit casts, they should enable the new lint (which should be `pedantic`).